docs(Contribution/handbook): updates contributions guides and handbook

abbeyhrt opened the pull request:
I updated the Contribution guides and did my best to make them a bit more beginner friendly for HacktoberFest. I ended up just chucking all the coding style stuff that was in the guide to the developer handbook since it felt more like noise in the guide than required information. I didn't check to make sure that info was still relevant so let me know if something needs to be updated! 

#### Testing / Reviewing

Read through the guide for typos or confusing sentences and check to make sure everything is still relevant.

abbeyhrt committed:
docs(Contribution/handbook): updated contributions guides and handbook)

abbeyhrt committed:
chore: remove comments and edit word choice

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
+## Prerequisites
 
-Set up your SSH Key GitHub account and install node.js 4 or higher.
+To contribute to Carbon, you'll need to install:
```
```suggestion
Before contributing to Carbon, you should make sure you have the following tools installed:
```

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
-- [Generating SSH Keys - GitHub](https://help.github.com/articles/generating-ssh-keys/)
-- [`nvm` (Node Version Manager)](https://github.com/creationix/nvm) to use the
-  `Node 6`.
+- [node(v10 or higher)](https://nodejs.org/en/download/)(you can download it
```
```suggestion
- [Node.js](https://nodejs.org/en/download/) v10 or above
```

I think we can drop the note around package manager (download link should be enough)

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
+- [node(v10 or higher)](https://nodejs.org/en/download/)(you can download it
+  here or follow their installation through a package manager
+  [here](https://nodejs.org/en/download/package-manager/))
+  - We recommend using
```
```suggestion
  - If you're on macOS, we recommend using [`nvm`](https://github.com/nvm-sh/nvm) to help manage different versions of Node.js 
```

mattrosno replied:
I use https://github.com/tj/n, but to each their own.

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
+  - We recommend using
+    [nvm](https://github.com/nvm-sh/nvm/blob/master/README.md) as your version
+    manager for Node.
+- [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
```
Should we share a link for `git`? This pathway is a hard one to predict given how many people use machines that have it pre-loaded 😬

mattrosno replied:
I agree. Plus, should be `Git`.

abbeyhrt replied:
Would you recommend just putting `git` or suggesting they check to see if they have it installed, telling them how, and if not supplying this link?

abbeyhrt replied:
Changed it to just Git with no link, should be good now 👍

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
+    [nvm](https://github.com/nvm-sh/nvm/blob/master/README.md) as your version
+    manager for Node.
+- [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [yarn](https://yarnpkg.com/en/docs/install#mac-stable)
```
```suggestion
- [yarn](https://yarnpkg.com/en/docs/install)
```

I think without the hash at the end it should automatically pick up the user's OS-specific install instructions 👍

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
 
-Contributing to carbon-components requires that you can run this repo locally on
-your computer.
+You will also need to have a GitHub account. Head over to
```
Do we need this section? Not sure what we're assuming when someone heads to this file, we might be able to assume they have a GitHub account but I'm not sure

abbeyhrt replied:
That's fair! I was mainly following what was previously there but considering this is stored in Github, it's likely they have an account.

mattrosno replied:
Not sure if the couple sentences here on GitHub and Yarn help: https://www.carbondesignsystem.com/tutorial/react/overview#yarn

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
+[github.com](https://github.com/) to make an account.
 
-## Coding style
+Finally you'll need a code editor to make changes to Carbon. There are many to
```
Should we highlight editor stuff? Same question as above, unsure what level we expect folks to be contributing from.

abbeyhrt replied:
I wasn't sure how beginner friendly we wanted to make this, I can remove this if you think it's too basic!

abbeyhrt replied:
I would like to keep the section about code editors. In my programming classes, it was the thing people had no idea about (and the class never talked about it) so i think another place that mentions what to use is better. Plus it's pretty easy to ignore if it doesn't apply to them

joshblack commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
-Then edits of `.scss` files in `/path/to/carbon-components/src` will be
-reflected to the development environment of your framework variant repository.
-You don't need to do anything in `carbon-components` side.
+With that all done, you're ready to start contributing to Carbon!
```
🎉

joshblack commented on docs/developer-handbook.md:
```diff
@@ -265,6 +296,319 @@ considered a chore that we are doing to keep things up-to-date.
 
 </details>
 
+## Coding style
```
Should we move this under the components part above?

abbeyhrt replied:
Does it only apply to carbon-components?

abbeyhrt committed:
Update .github/CONTRIBUTING.md

Co-Authored-By: Josh Black <josh@josh.black>

abbeyhrt committed:
Update .github/CONTRIBUTING.md

Co-Authored-By: Josh Black <josh@josh.black>

abbeyhrt committed:
Update .github/CONTRIBUTING.md

Co-Authored-By: Josh Black <josh@josh.black>

mattrosno commented on .github/CONTRIBUTING.md:
```diff
@@ -1,388 +1,98 @@
-- [`nvm` (Node Version Manager)](https://github.com/creationix/nvm) to use the
-  `Node 6`.
+- [Node.js](https://nodejs.org/en/download/) v10 or above
+  here or follow their installation through a package manager
```
Should first `here` be an anchor somewhere?

abbeyhrt committed:
Update .github/CONTRIBUTING.md

Co-Authored-By: Josh Black <josh@josh.black>

mattrosno commented on docs/developer-handbook.md:
```diff
@@ -19,6 +19,7 @@
   - [Body](#body)
   - [Footer](#footer)
   - [Examples](#examples)
+- [Coding Styles](#coding-styles)  
```
Should be `Coding style` and `#coding-style` to match the heading so the anchor link works.

mattrosno replied:
Actually, run `tocdoc` to generate this.

abbeyhrt committed:
chore: updates word choice

abbeyhrt committed:
Merge branch 'docs-update-contribution-guide' of https://github.com/abbeyhrt/my-carbon-fork into docs-update-contribution-guide

abbeyhrt committed:
chore: ran doctoc and updates contribution guide

abbeyhrt committed:
chore: updates content

joshblack committed:
Merge branch 'master' into docs-update-contribution-guide

tay1orjones commented on .github/CONTRIBUTING.md:
````diff
@@ -1,388 +1,97 @@
+
+```sh
 
 yarn dev
````
Running `yarn dev` from the root does nothing. Additionally, it would be helpful to outline that in addition to running `yarn`, you must also run `yarn build` before things will work. This is in the developer handbook, but would be helpful here as well.

abbeyhrt replied:
Good catch! Thanks for looking into this! I added more details around the build and starting the development server if you would like to check them out.

metonym replied:
Running `yarn build` at the root first should be emphasized. This would be especially helpful for those unfamiliar with the yarn/lerna monorepo set-up.

I had trouble setting up my environment because I hadn't run `yarn build` first. I was getting the "Can't resolve Package A, Package B..." when trying to run the `packages/react` Storybook even after a fresh clone and install.

abbeyhrt committed:
docs: adds info about running local server

abbeyhrt committed:
Merge branch 'docs-update-contribution-guide' of https://github.com/abbeyhrt/my-carbon-fork into docs-update-contribution-guide

jnm2377 commented:
Since you have a PR open, here's some content that might be nice to add. We drafted this as part of the Community/Contributions project last release cycle. These are more of just general contribution guidelines.

## updated code of conduct
```
# Carbon Code of Conduct

## TL;DR

- **Be respectful & understanding.** Not all of us will agree all the time, but
  disagreement is no excuse for poor behavior or poor manners. It is important
  that we resolve disagreements and differing views constructively.

* **Be welcoming.** We strive to be a community that welcomes and supports
  people of all backgrounds and identities.

## Our Pledge

In the interest of fostering an open and welcoming environment, we as
contributors and maintainers pledge to making participation in our project and
our community a harassment-free experience for everyone, regardless of age, body
size, disability, ethnicity, sex characteristics, gender identity and
expression, level of experience, education, socio-economic status, nationality,
personal appearance, race, religion, or sexual identity and orientation.

## Our Standards

Examples of behavior that contributes to creating a positive environment
include:

- Using welcoming and inclusive language
- Being respectful of differing viewpoints and experiences
- Gracefully accepting constructive criticism
- Focusing on what is best for the community
- Showing empathy towards other community members

Examples of unacceptable behavior by participants include:

- The use of sexualized language or imagery and unwelcome sexual attention or
  advances
- Trolling, insulting/derogatory comments, and personal or political attacks
- Degrading, demeaning or disrespectful comments
- Public or private harassment
- Publishing others' private information, such as a physical or electronic
  address, without explicit permission
- Other conduct which could reasonably be considered inappropriate in a
  professional setting

## Our Responsibilities

Project maintainers are responsible for clarifying the standards of acceptable
behavior and are expected to take appropriate and fair corrective action in
response to any instances of unacceptable behavior.

Project maintainers have the right and responsibility to remove, edit, or reject
comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, or to ban temporarily or permanently any
contributor for other behaviors that they deem inappropriate, threatening,
offensive, or harmful.

## Scope

This Code of Conduct applies within all project spaces, including _but not
limited to_ GitHub, e-mail and Slack. It also applies when an individual is
representing the project or its community in public spaces. Examples of
representing a project or community include using an official project e-mail
address, posting via an official social media account, or acting as an appointed
representative at an online or offline event. Representation of a project may be
further defined and clarified by project maintainers.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported by contacting the project team at carbon@us.ibm.com. All complaints
will be reviewed and investigated and will result in a response that is deemed
necessary and appropriate to the circumstances. The project team is obligated to
maintain confidentiality with regard to the reporter of an incident. Further
details of specific enforcement policies may be posted separately.

Project maintainers who do not follow or enforce the Code of Conduct in good
faith may face temporary or permanent repercussions as determined by other
members of the project's leadership.

## Slack

1.  Please, no `@here` or `@channel` unless it is a bug that is **breaking
    production code.** We have communities in many different time zones, and we
    would hate to disrupt someone not at the office.
2.  There's no need to notify the Carbon team that you have made an issue in
    Carbon's GitHub repositories. We check these several times a day, so we
    promise we'll address your issue as soon as we can.
3.  Reserve direct messages for sensitive discussions, such as privacy, legal,
    or HR reasons. Otherwise, we encourage posting in public channels so people
    with the same questions can benefit.
4.  If you have a pressing bug fix, it is best to make a PR directly to get your
    issues addressed.
5.  `#carbon-components`: This channel is for questions about Carbon Components
    only. It is not the place to ask general coding questions. Instead, use a
    dev community like [StackOverflow](https://stackoverflow.com/).
6.  `#carbon-design-system`: Please post any design questions with an in-context
    screen shot (i.e. screen shot of the whole UI you are designing) with
    background as to what you are trying to accomplish in this flow. We welcome
    in-progress work to get community design feedback as well.

## Attribution

This Code of Conduct is adapted from the
[Contributor Covenant](https://www.contributor-covenant.org), version 1.4,
available
[here](https://www.contributor-covenant.org/version/1/4/code-of-conduct.html).
For answers to common questions about this code of conduct, see the Contributor
Covenant [FAQ](https://www.contributor-covenant.org/faq).

<hr>

If you have suggestions to improve this Code of Conduct, please submit an issue
or PR.
```

## for `contribution.md`
```
# Carbon contribution model

Carbon Design System is an **open source** project at IBM. We pride ourselves in
open and inclusive design and development. If you're wondering more about our
contribution process, you're in the right place. First off, thank you for your
interest! This project is made possible not just by the core Carbon team, but
also by several community members who have invested their own time to give back
to the Carbon community.

## Code of conduct

We value all of our community members, and thus want to foster a positive
contributing environment. Please take a look at our
[code of conduct](./code_of_conduct.md) before engaging in our workspaces.

## FAQ

### Who can contribute?

Anyone! We mean it. The one and only requirement is you'll need a
[public GitHub account](https://github.com/join), as all our assets live on
GitHub.

- **Development:** If coding is your thing, you can help us by contributing bug
  fixes or community components. Checkout our
  [Developer Handbook](../developer-handbook.md) to get your dev environment set
  up, read up on our best practices and more.
- **Design:** Design contributions can vary from visual assets, UX interactions,
  motion design, Sketch kit bug fixes and more.
- **Content:** Our documentation is just as important as our design and code
  assets. Whether it's updating our current docs, or adding new
  [patterns](./patterns), anyone can contribute to our
  [website content](https://github.com/carbon-design-system/carbon-website).
- **Research:** Carbon is made up of developers and designers, but
  unfortunately, no dedicated researchers. If you're a researcher and have
  findings that you think could improve Carbon users' experience, you're in the
  right place. This kind of contribution is most effective if coupled with
  design and development forces, which would be presented in a GitHub issue and
  subsequent PR.

### What is the contribution process?

1. **Issue:** Check repo for an _existing_ issue related to your contribution
   first. If none exist, open a new issue. Be sure to check the right repo.
   (i.e. Don't open an issue for website documentation in the `carbon`
   monorepo.) We reserve the right to close any issues that haven't been filled
   out properly according to the issue template.
2. **Contributor License Agreement:** Before you can contribute any code, we
   need you to sign a Contributor License Agreement (CLA). Code doesn't just
   mean "components"; if you're contributing to our website docs, you're
   contributing code. ;)
3. **Development environment:** If you haven't already, fork and clone whichever
   repo you want to contribute to. Then, create a new branch and add your
   contribution in it. Checkout our
   [Developer Handbook](../developer-handbook.md) to read up on our best coding
   practices and proper commit messages.
4. **Pull request:** Submit a PR. Be sure to fill out the template properly.
5. **Approval:** Get PR approved by design and developers, or make any necessary
   changes for approval. This process may be quick or take a few iterations of
   feedback-update.
6. **Documentation:** After design and dev have approved and merged PR, update
   any website documentation if necessary. One of the best examples for this is
   if you're contributing to component work which has website documentation
   related to your contribution.

Here are some contribution quick tips:

- **Do** check repos for existing issues.
- **Do** fill out the required template for contributions entirely; this
  pertains to both issues and PRs.
- **Do** add or update tests for any contributions that require it.
- **Do** follow existing coding and writing styles.
- **Do** follow proper commit messages syntax.
- **Do not** branch off another branch.
- **Do not** include unrelated changes in the same PR.
- **Do not** create one massive PR if it can be broken up into smaller PRs.

### What projects can I contribute to?

- [`carbon-components`](https://github.com/carbon-design-system/carbon/tree/master/packages/components)
- [`carbon-components-react`](https://github.com/carbon-design-system/carbon/tree/master/packages/react)
- [`carbon-website`](https://github.com/carbon-design-system/carbon-website)
- [`gatsby-theme-carbon`](https://github.com/carbon-design-system/gatsby-theme-carbon)
- [`carbon-design-kit`](https://github.com/carbon-design-system/carbon-design-kit)
- [`@carbon/elements`](https://github.com/carbon-design-system/carbon/tree/master/packages/elements)
- [`@carbon/type`](https://github.com/carbon-design-system/carbon/tree/master/packages/type)
- [`@carbon/colors`](https://github.com/carbon-design-system/carbon/tree/master/packages/colors)
- [`@carbon/grid`](https://github.com/carbon-design-system/carbon/tree/master/packages/grid)
- [`@carbon/icons`](https://github.com/carbon-design-system/carbon/tree/master/packages/icons)
- [`@carbon/layout`](https://github.com/carbon-design-system/carbon/tree/master/packages/layout)
- [`@carbon/motion`](https://github.com/carbon-design-system/carbon/tree/master/packages/motion)
- [`@carbon/themes`](https://github.com/carbon-design-system/carbon/tree/master/packages/themes)
- [`@carbon/icons-react`](https://github.com/carbon-design-system/carbon/tree/master/packages/icons-react)

And more! You can view a complete list of our packages
[here](https://github.com/carbon-design-system/carbon/tree/master/packages).

### Besides some of the obvious contributions mentioned above, how else can I contribute?

Great question! Contribution does not require creating or maintaining our
assets. Here are some other ways you can contribute, which don't require much
work:

- **Join our slack community and interact with other users.** We have hundreds
  of users world wide, and quite a small team in comparison. While we try our
  best to answer questions on slack, it's not always possible to answer
  everything. One of the easiest ways to help us it to jump in on slack
  conversations if there's something you know the answer to! Some of our most
  popular slack channels include `#carbon-components`, `#carbon-react`,
  `#carbon-design-system`, `#carbon-ng`, `#carbon-vue`, `#carbon-announcements`,
  `#carbon-community-components`.
- **Report bugs.** Even if you don't have the time to contribute a bug fix,
  opening an issue alone makes a big difference! Be sure to completely fill out
  the issue template to best help us understand what is going wrong.

### If I'm contributing code, am I required to contribute it for all frameworks?

If you've been working with Carbon for a while, you know that we have code
assets in vanilla JS, React, Vue and Angular. While the core team only maintains
the vanilla and React components, we work closely with the Vue and Angular teams
to maintain parity. The core team also develops with a React first approach,
which means that our vanilla components will never be ahead of React, though it
may not be true for the other way around. If you're contributing a bug fix in a
vanilla component which also exists in the React version, you can either
contribute a fix for the React version as well or open an issue so that we can
update the React component accordingly.
```

abbeyhrt commented:
@jnm2377 nice addition! Would we want this in the contribution guide or I can link out in the guide and put this in the handbook?

abbeyhrt committed:
Merge branch 'master' into docs-update-contribution-guide

jnm2377 commented:
@abbeyhrt I think it makes more sense to have it in the contribution guide/code of conduct since it seems that the handbook is more technical and this is more general.

abbeyhrt committed:
docs: adds code of conduct and contribution guidelines

abbeyhrt committed:
Merge branch 'docs-update-contribution-guide' of https://github.com/abbeyhrt/my-carbon-fork into docs-update-contribution-guide

abbeyhrt committed:
docs: add coe of conduct link to guide

abbeyhrt commented:
@jnm2377 I added the Code of Conduct to the developer handbook and linked to it in the guide and added the contribution model to the guide! I just split it up a bit, put the `## Carbon Contribution Model` and ` Code of conduct` section at the top and the FAQs at the bottom of the guide! Let me know if you want me to change any of that 🙂

abbeyhrt commented on .github/CONTRIBUTING.md:
```diff
@@ -1,478 +1,177 @@
 # Contributing
 
-## Requirements
+## Carbon contribution model
```
```suggestion
## Our contribution model
```

abbeyhrt committed:
Update .github/CONTRIBUTING.md

jnm2377 commented:
@abbeyhrt contribution doc looks great!
Do you wanna just update the code of conduct doc instead of the handbook though (for the conduct guidelines I shared)? 

https://github.com/carbon-design-system/carbon/blob/master/.github/CODE_OF_CONDUCT.md

abbeyhrt commented:
@jnm2377  yeah, i didn't realize we already had a code of conduct document already 🤦‍♀ my bad lol

abbeyhrt committed:
docs: removed code of conduct from handbook, updates code of conduct

abbeyhrt committed:
docs: updates yarn install command

abbeyhrt committed:
chore: merge in changes

jnm2377 reviewed (approved):
🎉
